src/components/ec-btn/ec-btn.spec.js

```diff
@@ -0,0 +1,200 @@
+        to: 'trade-finance',
+        tag: 'ebury',
+      },
+      {
+        stubs: ['ebury'],
+      });
+
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a disabled button when disabled is set to true', () => {
+      const wrapper = mountBtn({
+        disabled: true,
+      });
+
+      expect(wrapper.attributes().disabled).toBe('disabled');
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with an icon when the icon prop is defined', () => {
+      const wrapper = mountBtn({
+        icon: 'simple-check',
+      });
+
+      expect(wrapper.find('.ec-btn__icon').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with an outline when "outline" prop is set to true', () => {
+      const wrapper = mountBtn({
+        outline: true,
+      });
+
+      expect(wrapper.find('.ec-btn--outline').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a full width button when "fullWidth" prop is set to true', () => {
+      const wrapper = mountBtn({
+        fullWidth: true,
+      });
+
+      expect(wrapper.find('.ec-btn--full-width').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it.each(['primary', 'secondary', 'success', 'error', 'warning'])('should render a button with type "%s" when type prop is set', (type) => {
+      const wrapper = mountBtn({
+        type,
+      });
+
+      expect(wrapper.find(`.ec-btn--${type}`).exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a reverse button when reverse is set to true', () => {
+      const wrapper = mountBtn({
+        reverse: true,
+      });
+
+      expect(wrapper.find('.ec-btn--reverse').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with a spinner loader if loading is set to true', () => {
+      const wrapper = mountBtn({
+        loading: true,
+      });
+
+      expect(wrapper.attributes().disabled).toBe('disabled');
+      expect(wrapper.find('.ec-btn__spinner').exists()).toBe(true);
+      expect(wrapper.element).toMatchSnapshot();
+    });
+
+    it('should render a button with type submit if submit is set to true', () => {
+      const wrapper = mountBtn({
+        submit: true,
+      });
+
+      expect(wrapper.attributes().type).toBe('submit');
+      expect(wrapper.element).toMatchSnapshot();
+    });
+  });
+
+  describe(':slots', () => {
+    it('should render a button with "Click me!" text', () => {
+      const wrapper = mountBtn(
+        {},
+        {
+          slots: {
+            default: 'Click Me!',
+          },
+        },
+      );
+      expect(wrapper.find('.ec-btn__spinner').exists()).toBe(false);
+      expect(wrapper.findByDataTest('ec-btn__text-loader-slot').exists()).toBe(false);
```

Comment:
You can user `findByDataTest` for other tests too. Many `.find()` calls can be replaced with `.classes()`,  rest of them can be done via `findByDataTest()`. It's more preferable so we do not tie-up our classes to the tests.